src/main/java/org/broadinstitute/hellbender/tools/dataflow/transforms/bqsr/BaseRecalibratorFn.java

```diff
@@ -81,167 +59,118 @@
-    }
+        baq = new BAQ(BRAC.BAQGOP); // setup the BAQ object with the provided gap open penalty
 
-    private GATKRead setDefaultBaseQualities(GATKRead read) {
-        // if we are using default quals, check if we need them, and add if necessary.
-        // 1. we need if reads are lacking or have incomplete quality scores
-        // 2. we add if defaultBaseQualities has a positive value
-        if (BRAC.defaultBaseQualities < 0) {
-            return read;
-        }
-        byte reads[] = read.getBases();
-        byte quals[] = read.getBaseQualities();
-        if (quals == null || quals.length < reads.length) {
-            byte new_quals[] = new byte[reads.length];
-            Arrays.fill(new_quals, BRAC.defaultBaseQualities);
-            read.setBaseQualities(new_quals);
-        }
-        return read;
-    }
+        if (BRAC.RAC.FORCE_PLATFORM != null)
+            BRAC.RAC.DEFAULT_PLATFORM = BRAC.RAC.FORCE_PLATFORM;
 
-    public RecalibrationTables getTables() {
-        return recalibrationEngine.getRecalibrationTables();
-    }
+        covariates = new StandardCovariateList(BRAC.RAC, header);
 
-    /**
-     * Returns the requested covariates, possibly mutated during the execution.
-     */
-    public StandardCovariateList getRequestedCovariates() {
-        return this.covariates;
-    }
+        int numReadGroups = header.getReadGroups().size();
 
-    /**
-     * @param rt recalibration tables
-     * @return the quantization information
-     */
-    public QuantizationInfo getQuantizationInfo(RecalibrationTables rt) {
-        return new QuantizationInfo(rt, BRAC.RAC.QUANTIZING_LEVELS);
-    }
+        recalibrationEngine = new RecalibrationEngine(covariates, numReadGroups);
 
-    private ReadTransformer makeReadTransform() {
-        ReadTransformer f0 = BaseRecalibratorWorker::consolidateCigar;
+        minimumQToUse = BRAC.PRESERVE_QSCORES_LESS_THAN;
 
-        ReadTransformer f =
-                f0.andThen(this::setDefaultBaseQualities)
-                        .andThen(this::resetOriginalBaseQualities)
-                        .andThen(ReadClipper::hardClipAdaptorSequence)
-                        .andThen(ReadClipper::hardClipSoftClippedBases);
 
-        return f;
     }
 
-
     /**
      * For each read at this locus get the various covariate values and increment that location in the map based on
      * whether or not the base matches the reference at this particular location.
      * The "knownLocations" list doesn't need to be complete but it must includes all those that overlap with the read.
      */
-    public void apply(GATKRead originalRead, final ReferenceContext ref, List<? extends Locatable> knownLocations) {
+    private void apply(GATKRead originalRead, final ReferenceContext ref, final ReferenceDataSource refDS, Iterable<? extends Locatable> knownLocations) {
         ReadTransformer transform = makeReadTransform();
         final GATKRead read = transform.apply(originalRead);
```

Comment:
Add a note that the read transformers may change the position of the read by clipping bases off the ends.